src/Core/Form/IdentifiableObject/CommandBuilder/Product/SEOCommandsBuilder.php

```diff
@@ -61,6 +63,18 @@ public function buildCommands(ProductId $productId, array $formData): array
             $command->setRedirectOption($redirectionData['type'], $targetId);
         }
 
-        return [$command];
+        if (!empty($seoData['tags'])) {
```

Comment:
```suggestion
        if (!empty($seoData['tags']) && is_array($seoData['tags'])) {
```

For me, nothing says it's an array

Reply:
Just in case the tags are an iterable object:
```suggestion
        if (!empty($seoData['tags']) && is_iterable($seoData['tags'])) {
```

Reply:
But we don't usually test the type of array fields in all our builders, we kind of expect that the format from the form matches our expectation Not all fields are mandatory to be here, so we check their presence, but when they are present they should be in the right format

It's already the case for all the translatable fields for example, `$seoData['meta_title']` is an array of values indexed by language ID, we inject the value in the set method without pre-checking because if it's not it will break (because of strong typing) and we want it to break Unexpected format should result in an error not be ignored

But in this case, you're right since the value is interpreted and reformatted before being injected in the command constructor we should check it, but I'll throw an exception instead if the field is not an array

And I don't think `is_iterable` is necessary, all our data are flattened array so `is_array` is enough Using `is_iterable` would sort of imply that another type than array is usable It's not expected so I prefer to insist on the array

Reply:
Your point makes sense... and I just can't trust arrays :D

Reply:
I approve @saulaski